### DISCUSSION TITLE
A UQTk Install Problem

### DISCUSSION THREAD

**Discussion by qhyxff** [sandialabs/UQTk#14] [Q&A] [answered]
Hello! I'm a green hand to UQTk; I've download newest UQTk and Sundials package from website. When I build UQTk the system always remind me of something wrong with cvcode, which seems to be a problem with UQTk using Sundials. However, Sundials can run normally on my Ubuntu. Is there anyone know whats wrong with that?
![Screenshot 2024-10-24 161043](https://github.com/user-attachments/assets/8af7d56a-da97-410b-a290-c0d6bef6f106)

**Comment by bjdebus** [comment]
Hi @qhyxff , thanks for reaching out. Sorry to hear you are having compilation problems with UQTk. Can you tell me what version of Sundials you are using so I can try to replicate the error on my end?

In the meantime, as a workaround,  you can also have UQTk download Sundials as a submodule and then UQTk should pull the version of Sundials that it is compatible with. This works well if you do not already have Sundials installed elsewhere on your system (as you may not want two versions of Sundials laying around). Thanks! Bert

**Reply by qhyxff** [reply]
Thanks for replying and helping!

My sundials are newest version from official website V7.1.1. I've search entire installed Sundials program folders and couldn't find cvode_direct.h.

By the way, I've also tried to let UQTk download Sundials; However, the wrong message often came when I run bash file to build environment:
![Screenshot 2024-10-25 095540](https://github.com/user-attachments/assets/45c026a4-6729-45bf-8de0-93c4b5d75bf5)
It seems something also wrong for downloading this package. I've looked at the cmakelists in /dep and found that maybe this part of code cannot work on my ubuntu:
![1](https://github.com/user-attachments/assets/2fb83465-1fb5-4a84-8432-0915baef782f)

**Reply by bjdebus** [reply]
Thanks @qhyxff for this additional information. That gives me something to work with. How urgent is this issue for you? It may take me a few days to dig into this. However, if this is urgent, I can try to get to it sooner. Alternatively, if you tell me what you want to use UQTk for, I can maybe give you a workaround that does not depend on SunDials. Either way, I will figure out what is happening here. Regards, Bert

**Reply by qhyxff** [reply]
Thanks for your reply! It's not urgent right now because I just use a previous version two years ago from my colleague's Ubuntu and it can make well and install; However, the newest version still run bad on my Ubuntu.

I'll use UQTk for some turbulence model uncertainty quanlification work during CFD process

**Reply by bjdebus** [reply]
Hi @qhyxff , 
Just to let you know, I have not forgotten about this. Tried to work on it last week, but ran into a problem. Will let you know when I have an update.
Regards,
Bert

**Comment by bjdebus** [comment, marked as answer]
@qhyxff I just pushed UQTk version 3.1.5 to github. In this release, UQTk has been made compatible with sundials 7.1.1.
Can you try to compile this latest version on your machine and let me know if it works?
Many thanks for your patience!

**Reply by qhyxff** [reply]
Thanks a lot! I'll try the latest version and let you know if something goes wrong

**Reply by anufagbemi** [reply]
Hi @qhyxff, I don't know if you were able to solve the problem yet, but I did get it to install by doing the following:

$ cmake  -DCMAKE_SUNDIALS_DIR=$HOME/uncertainty_quantification/UQTK/Dependencies/build-sundial_711   ../UQTk/

Setting the path to the build manually was the only way I could get it to find sundial. You might also need to create a symbolic link for lgfortran for the package to build successfully, though not the best route. My system couldn't find it for some reason although it is installed.
